web/packages/teleterm/src/ui/services/connectionTracker/types.ts

```diff
@@ -39,6 +39,13 @@ export interface TrackedGatewayConnection extends TrackedConnectionBase {
   targetSubresourceName?: string;
 }
 
+export interface TrackedGatewayKubeConnection extends TrackedConnectionBase {
```

Comment:
The implementation of different bits of `connectionTracker` seems correct. However, I don't think we want to add a new type here, instead we should opt to reuse `TrackedKubeConnection` and make it work with the new doc type.

I assume that after adding kube proxies, we'll want to make it so that when the user updates the app, all of their existing kube tabs will now use the kube gateway.